docs/getting-started-guides/docker-multinode.md

```diff
@@ -74,54 +81,146 @@ This pattern is necessary because the `flannel` daemon is responsible for settin
+
+```sh
+export NODES="vcap@10.10.102.150 vcap@10.10.102.152"
+export MASTER="vcap@10.10.102.150"
+export KUBERNETES_PROVIDER=docker-cluster ./kube-up.sh
+```
+
+> Check `cluster/docker-cluster/config-default.sh` for more supported ENVs
+
+If all things goes right, you will see the below message from console indicating the k8s is up.
+
+```console
+Deploy Complete!
+... calling validate-cluster 
+... Everything is OK! 
+```
+
+### Add another Node into cluster
+
+Adding a Node to existing cluster is quite easy, just set `NODE_ONLY` to clarify you want to provision Node only:
+
+```sh
+export NODE_ONLY=yes
+export NODES="vcap@10.10.102.153"
+export MASTER="vcap@10.10.102.150"
+```
+
+## Test it out
+
+On every node, you can see there're two containers running by `docker ps`:
+
+```console
+kube_in_docker_proxy_xxx
+kube_in_docker_kubelet_xxx
+```
+
+And on Master node, you can see extra master containers running:
+
+```console
+k8s_scheduler.xxx
+k8s_apiserver.xxx
+k8s_controller-manager.xxx
+```
+
+> Currently, we assume 'Master only' node is meaningless, but please fire up issue if you want that, we can set `-runonce=false` for kubelet on the Master node
+
+As we use `hyperkube` image to run k8s, we **do not** need to compile binaries, please download and extract `kubectl` binary from [releases page](https://github.com/kubernetes/kubernetes/releases).
+
+At last, use `$ kubectl get nodes` to see if all of your nodes are ready.
+
+```console
+$ kubectl get nodes
+NAME            LABELS                                 STATUS
+10.10.102.150   kubernetes.io/hostname=10.10.103.150   Ready
+10.10.102.153   kubernetes.io/hostname=10.10.102.153   Ready
+```
+
+### Deploy a DNS
+
+See [here](docker-multinode/deployDNS.md) for instructions.
+
+Then you can run Kubernetes [guest-example](../../examples/guestbook/) to build a redis backend cluster on the k8s．
+
+## Customize
 
-## Master Node
+One of the biggest benefits of using Docker to run Kubernetes is users can customize the cluster freely before you running `kube-up.sh`:
 
-The first step in the process is to initialize the master node.
+### master
 
-Clone the Kubernetes repo, and run [master.sh](docker-multinode/master.sh) on the master machine with root:
+You need to enable **comstomized master mode** before deploy:
 
 ```sh
-cd kubernetes/docs/getting-started-guides/docker-multinode/
-./master.sh
+export MASTER_CONF=yes
 ```
 
-`Master done!`
+The configure file of Master locates in `docker-cluster/kube-config/master-multi.json`, which will be mounted as volume for Master Pod to comsume, you can modify it freely **before deploying**.
 
-See [here](docker-multinode/master.md) for detailed instructions explanation.
+In this mode, you can even change the configuration of Master after the deployment has done, see:
 
-## Adding a worker node
+1. Login the Master node
+2. Change the content in `~/docker-cluster/kube-config/master-multi.json`
+3. Restart the affected master containers
```

Comment:
Actually this step is not needed. Kubelet will notice file change and will restarts pods on its own.